model/task_lifecycle.go

```diff
@@ -306,16 +308,9 @@ func DeactivatePreviousTasks(t *task.Task, caller string) error {
 			continue
 		}
 
-		err = SetActiveState(&t, caller, false)
-		if err != nil {
+		if err = SetActiveState(&t, caller, false); err != nil {
 			return err
 		}
-		// update the cached version of the task, in its build document to be deactivated
-		if !t.IsPartOfDisplay() {
```

Comment:
Why did you remove this?

Reply:
I was trying to consolidate the places where we update the build cache tasks' activation to the places where we modify the actual tasks' activations. 
Now SetActiveState handles setting the build cache itself. 

I really would have wanted to do it inside the task package (in the functions that (de)activate tasks) but it made an import cycle (because the build package imports the task package). Instead, I settled for updating both the task and build cache from the model package (task_lifecycle and lifecycle).